### PR TITLE
📃 [DOCS]  Add explanation section for anomaly tasks.

### PR DESCRIPTION
**Pull request by samet-akcay**
# Description
- This PR adds the explanation section for anomaly tasks. 

# Motivation
- This is a documentation requirement as part of the v1.0.0 release

# Limitations
- The scope of the PR is limited to the explanation section only. A follow up PR for the tutorial section will be created by @ashwinvaidya17 later on.

### CODE REVIEW

File: docs/source/guide/explanation/algorithms/anomaly/index.rst [hunk 1]
```diff
@@ -1,10 +1,123 @@
...
+   │       ├── 000.png
+   │       └── ...
+   └── train
+      └── good
+         ├── 000.png
+         └── ...
+
+Future OTX releases will support other benchmark datasets such as Amazon's `Visual Anomaly (VisA) <https://github.com/amazon-science/spot-diff#data-download>`_ dataset. Meanwhile, you can use the `MVTec AD dataset <https://www.mvtec.com/company/research/datasets/mvtec-ad/>`_ to train and test anomaly detection models, or use MVTec dataset structure to train and test anomaly detection models on your own dataset.
+
+Models
```
Comment (kprokofi) [inline docs/source/guide/explanation/algorithms/anomaly/index.rst:78]: We need to add pivot table with templates, params, FLOPs and links to template in this section. Also, pivot table with metrics on few academic datasets is also useful to expose (if we have it now, if not let's skip it for this release)

File: docs/source/guide/explanation/algorithms/anomaly/index.rst [hunk 2]
```diff
@@ -1,10 +1,123 @@
...
+
+.. _fig-anomaly-tasks:
+
+.. figure:: ../../../../../utils/images/anomaly_tasks.png
+   :width: 600
+   :align: center
+   :alt: Anomaly Task Types
+
+   Anomaly task types (a) A normal image used during training. (b) An image-level prediction of an anomaly by anomaly classification task (c) A box-level prediction of an anomaly by an anomaly detection task (d) A pixel-level prediction of an anomaly by anomaly segmentation task
+
```
Comment (kprokofi) [inline docs/source/guide/explanation/algorithms/anomaly/index.rst:44]: We also add description of the components for the specific algorithm. I mean, what is the data augmentations, optimizer, learning rate scheduler, other training techniques. Is it possible to add it there also?

**Review by kprokofi — CHANGES_REQUESTED** [review]
Thanks for this PR! I left comments. Overall, the structure for anomaly for now quite different from other tasks. Could you add some more description for algo components and pivot tables with templates summary and metrics on academic datasets (if available)

Also, is there not so much info to separate tasks for anomaly? For example like we did for classification: https://github.com/openvinotoolkit/training_extensions/tree/develop/docs/source/guide/explanation/algorithms/classification

File: docs/source/guide/explanation/algorithms/anomaly/index.rst [hunk 3]
```diff
@@ -1,10 +1,123 @@
...
+Task Types
+**********
+OTX supports the following anomaly task types:
+
+* Anomaly Classification - (Image-level detection)
+* Anomaly Detection - (Box-level localization)
+* Anomaly Segmentation - (Pixel-level localization)
+
+.. note::
+   Only normal images are used to train all anomaly detection algorithms. The distinction between the tasks is the way in which the testing phase predicts or localises anomalies. In anomaly classification, for instance, anomalies are predicted at the image level; in anomaly detection, anomalies are localised using a bounding box; and in anomaly segmentation, anomalies are localised using a mask.
```
Comment (djdameln) [inline docs/source/guide/explanation/algorithms/anomaly/index.rst]: We could add here that most anomaly models support all three task types

File: docs/source/guide/explanation/algorithms/anomaly/index.rst [hunk 4]
```diff
@@ -1,10 +1,123 @@
...
+
+
+Anomaly Classification
+----------------------
+Anomaly classification is the task of predicting normal and abnormal images at the image level. As noted above, a model is trained on only normal images. During the testing phase, the model predicts an anomaly score indicating the likelihood of an image being abnormal. The threshold for anomaly classification is either set by the user or adaptively tuned by OTX. An image is classified as abnormal if the anomaly score is above the threshold.
```
Comment (djdameln) [inline docs/source/guide/explanation/algorithms/anomaly/index.rst]: ~~OTX~~ OpenVINO Training Extensions

File: docs/source/guide/explanation/algorithms/anomaly/index.rst [hunk 5]
```diff
@@ -1,10 +1,123 @@
...
+.. figure:: ../../../../../utils/images/anomaly_tasks.png
+   :width: 600
+   :align: center
+   :alt: Anomaly Task Types
+
+   Anomaly task types (a) A normal image used during training. (b) An image-level prediction of an anomaly by anomaly classification task (c) A box-level prediction of an anomaly by an anomaly detection task (d) A pixel-level prediction of an anomaly by anomaly segmentation task
+
+Dataset Format
+**************
+At the moment, OTX anomaly tasks support the MVTec AD dataset format, which is one of the most popular formats for detecting anomalies.
```
Comment (djdameln) [inline docs/source/guide/explanation/algorithms/anomaly/index.rst]: ~~OTX~~ OpenVINO Training Extensions

File: docs/source/guide/explanation/algorithms/anomaly/index.rst [hunk 6]
```diff
@@ -1,10 +1,123 @@
...
+   │   │   └── ...
+   │   └── good
+   │       ├── 000.png
+   │       └── ...
+   └── train
+      └── good
+         ├── 000.png
+         └── ...
+
+Future OTX releases will support other benchmark datasets such as Amazon's `Visual Anomaly (VisA) <https://github.com/amazon-science/spot-diff#data-download>`_ dataset. Meanwhile, you can use the `MVTec AD dataset <https://www.mvtec.com/company/research/datasets/mvtec-ad/>`_ to train and test anomaly detection models, or use MVTec dataset structure to train and test anomaly detection models on your own dataset.
```
Comment (djdameln) [inline docs/source/guide/explanation/algorithms/anomaly/index.rst]: ~~OTX~~ OpenVINO Training Extensions

File: docs/source/guide/explanation/algorithms/anomaly/index.rst (hunk 5 above)
Comment (djdameln) [inline docs/source/guide/explanation/algorithms/anomaly/index.rst]: It might be a bit confusing that the tutorial uses a different format than MVTec. I think we should either change this description, or use an MVTec dataset in the tutorial.

File: docs/source/guide/explanation/algorithms/anomaly/index.rst (hunk 4 above)
Comment (djdameln) [reply docs/source/guide/explanation/algorithms/anomaly/index.rst]: Maybe we could provide a brief 1 sentence explanation how the adaptive threshold value is determined.

File: docs/source/guide/explanation/algorithms/anomaly/index.rst [hunk 7]
```diff
@@ -1,10 +1,123 @@
...
 
+This section examines the solutions for anomaly detection offered by the OpenVINO Training Extensions library.
 
-   anomaly_classification
-   anomaly_detection
-   anomaly_segmentation
+
+Task Types
+**********
+OpenVINO Trainign Extensions supports the following anomaly task types:
```
Comment (ashwinvaidya17) [inline docs/source/guide/explanation/algorithms/anomaly/index.rst]:
```suggestion
OpenVINO Training Extensions supports the following anomaly task types:
```

File: docs/source/guide/explanation/algorithms/anomaly/index.rst (hunk 1 above)
Comment (djdameln) [reply docs/source/guide/explanation/algorithms/anomaly/index.rst:78]: I added the templates table. For the metrics table let's wait until the next release. The current benchmarking results that we have available are outdated.

**Comment by djdameln** [comment]
@kprokofi 
> Also, is there not so much info to separate tasks for anomaly? For example like we did for classification: https://github.com/openvinotoolkit/training_extensions/tree/develop/docs/source/guide/explanation/algorithms/classification

The different anomaly tasks (anomaly classification, anomaly detection, anomaly segmentation) are based on the same models under the hood. The difference lies in the post-processing methods that are applied to the raw (pixel-level) predictions that are generated by the models. This is why we added a single page for the anomaly tasks combined. There is just too much overlap between the tasks to fill separate explanations for the individual tasks. I updated the note under Task Types to explain this a bit more clearly to the reader.

File: docs/source/guide/explanation/algorithms/anomaly/index.rst (hunk 2 above)
Comment (djdameln) [reply docs/source/guide/explanation/algorithms/anomaly/index.rst:44]: We added the descriptions of the training parameters for the individual models. (except for PADIM, because its training does not involve fine-tuning of network weights).

File: docs/source/guide/explanation/algorithms/anomaly/index.rst [hunk 8]
```diff
@@ -1,10 +1,162 @@
...
+
+Training Parameters
+~~~~~~~~~~~~~~~~~~~~
+
+Since STFPM trains the student network, we use the following parameters for its training:
+
+- ``Backbone``: The default backbone is ``ResNet18``. You can also use ``Wide ResNet50``.
+- ``Loss``: Loss is computed as the mean squared error between the student and teacher feature pyramids. The default loss is ``MSE`` and cannot be changed.
+- ``Optimizer``: The default optimizer is ``SGD`` and cannot be changed. It uses the following parameters that can be changed from the UI.
```
Comment (kprokofi) [inline docs/source/guide/explanation/algorithms/anomaly/index.rst]:
```suggestion
- ``Optimizer``: The default optimizer is ``SGD`` and cannot be changed. It uses the following parameters that can be changed:
```
Since this documentation for the OTX, not for the Geti - we don't have UI

**Review by kprokofi — COMMENTED** [review]
Please, delete UI mention and I will approve it, thank you for the efforts and great explanation section!

**Review by djdameln — APPROVED** [review]

**Review by goodsong81 — APPROVED** [review]
Thanks! Looks great :)
